client/host_queue.go

```diff
@@ -456,3 +450,23 @@ func (q *queue) Close() {
 	close(q.drainIn)
 	q.Unlock()
 }
+
+// errors
+
+type errQueueNotOpen string
```

Comment:
Instead of implementing the `error` interface, might it not be better to just make these methods create errors using the standard `fmt.Errorf`? 

e.g. 
```go
func newErrQueueNotOpen(hostID string) error {
   return fmt.Errorf("host operation queue not open for host: %s", hostID)
}

func newErrQueueUnknownOperation(hostID string) {
    return fmt.Errorf("host operation queue received unknown operation for host: %s", hostID)
}
```